engine/_classes/class.bertagallery.php

```diff
@@ -91,25 +92,47 @@ public static function getHTML($imgs, $mediaFolderName, $galleryType, $isAdminMo
                     break;
             }
 
-            $dimensions = ' style="width: ' . $firstImageWidth . 'px;' . ($galleryType !== 'slideshow' ? 'height: ' . $firstImageHeight . 'px;' : '') . '"';
-            $strOut = '<div class="xGalleryContainer xGalleryHasImages xGalleryType-' . $galleryType . $specificClasses . '"'. ($galleryFullScreen ? ' data-fullscreen="1"' : '') .'>';
-            $strOut .= "<div class=\"xGallery\"" . $dimensions . ($rowGalleryPadding ? ' xRowGalleryPadding="' . $rowGalleryPadding . '"' : '') . '>';
+            $galleryContent = '';
+            $widestImage = $firstImageWidth;
 
             if ($galleryType == 'slideshow') {
                 $slides = [];
                 foreach ($imgs as $img) {
                     if ($img['@attributes']['type'] == 'image') {
-                        list($slideHTML) = BertaGallery::getImageHTML($img, $mediaFolderName, $isAdminMode, $sizeRatio, $imageTargetWidth, $imageTargetHeight);
+                        list($slideHTML, $slideWidth) = BertaGallery::getImageHTML($img, $mediaFolderName, $isAdminMode, $sizeRatio, $imageTargetWidth, $imageTargetHeight);
+                        if ($widestImage < $slideWidth) {
+                            $widestImage = $slideWidth;
+                        }
                     } else {
-                        $slideHTML = BertaGallery::getVideoHTML($img, $mediaFolderName, $isAdminMode, $sizeRatio, $imageTargetWidth, $imageTargetHeight);
+                        list($slideHTML, $slideWidth) = BertaGallery::getVideoHTML($img, $mediaFolderName, $isAdminMode, $sizeRatio, $imageTargetWidth, $imageTargetHeight);
+                        if ($widestImage < $slideWidth) {
+                            $widestImage = $slideWidth;
+                        }
                     }
                     $slides[] = $slideHTML;
                 }
-                $strOut .= BertaGallery::getSlideshowHTML($slides);
+                $galleryContent .= BertaGallery::getSlideshowHTML($slides);
             } else {
-                $strOut .= $firstImageHTML;
+                $galleryContent .= $firstImageHTML;
             }
 
+            // Set slideshow gallery width by widest slide if
+            // - gallery type is slideshow and
+            // - current template is messy and gallery setting `galleryWidthByWidestSlide` is ON
+
+            // @TODO consider to set slideshow width by widest slide for all other temaplates by default
+
+            $isMessyTemplate = strpos($berta->settings->get('template', 'template'), 'messy') === 0;
```

Comment:
We should probably have helper functions, that can check the current template given the settings. That would simplify any kind of centralized template update or fix to template type checking.